src/bom/chainsop.rs

```diff
@@ -0,0 +1,872 @@
+                            .collect();
+                        return Ok(SubProcFile::StaticOutputFile(fpath.clone()));
+                    }
+                    NamedFile::GlobIn(dpath, glob) => {
+                        let mut bc_glob = String::new();
+                        bc_glob.push_str(&OsString::from(dpath).into_string().unwrap());
+                        bc_glob.push_str("/");
+                        bc_glob.push_str(glob);
+                        let bc_files = glob::glob(&bc_glob)?
+                            .filter_map(Result::ok)
+                            .map(|x| OsString::from(x).into_string().unwrap())
+                            .collect::<Vec<String>>()
+                            .join(",")
+                            ;
+                        args.push(OsString::from(bc_files))
+                    }
+                }
+        }
+        Ok(SubProcFile::NoOutputFile)
+    }
+
+    // Alternative to the setup_file function which instead uses the specified
+    // file instead of the existing specification, modifying the args in the
+    // appropriate manner to insert the file reference.
+    fn setup_file_override(&self,
+                           ovrf: &PathBuf,
+                           mut args: Vec<OsString>)
+                           -> Vec<OsString>
+    {
+        match &self {
+            FileSpec::Unneeded => args,
+            FileSpec::Append(_) => { args.push(ovrf.into()); args },
+            FileSpec::Option(flg, _) => {
+                args.push(OsString::from(flg));
+                args.push(ovrf.into());
+                args
+            },
+            FileSpec::Replace(pat, _) =>
+                args.into_iter()
+                .map(|arg| replace(pat, &ovrf.into(), &arg))
+                .collect()
+        }
+    }
+}
+
+
+// ----------------------------------------------------------------------
+// Single sub-process operation management
+
+/// This structure represents a single command to run as a sub-process, the
+/// command's arguments, and the input and output files for that sub-process.
+/// The structure itself is public but the fields are private
+/// (i.e. implementation specific); the impl section below defines the visible
+/// operations that can be performed on this structure.
+#[derive(Debug)]
+pub struct SubProcOperation {
+    cmd : Operation,
+    args : Vec<OsString>,
+    inp_file : FileSpec,
+    out_file : FileSpec,
+    in_dir : Option<PathBuf>,
+}
+
+enum Operation {
+    /// Name of executable to invoke in subprocess
+    Execute(OsString),
+
+    /// Local function to call instead of executing a subprocess.  The first
+    /// argument is the current working directory, the second is the argument
+    /// vector (the input and output files will be part of the argument vector as
+    /// determined by their corresponding FileSpec).
+    Call(Box<dyn Fn(&Path, Vec<OsString>) -> anyhow::Result<()>>)
```

Comment:
Could you add an example in the documentation describing why one might want this and what it is for?

Reply:
Done.